pkg/client/actionclient.go

```diff
@@ -125,6 +134,39 @@ func NewActionClientGetter(acg ActionConfigGetter, opts ...ActionClientGetterOpt
 	return actionClientGetter, nil
 }
 
+type Preflight interface {
+	Run(string, *release.Release) error
```

Comment:
Instead of passing a string representing whether it's an install/upgrade/uninstall, could we have a struct that implements separate functions for each phase as a no-op by default where a user can provide "overrides" for just the preflight phases they care about. Basically something something similar to controller-runtime's [handler.Funcs](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/handler#Funcs)